src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -169,6 +169,28 @@ export class MockHttpSocket extends MockSocket {
       }
     }
 
+    // Forward TLS Socket properties onto this Socket instance
+    // in the case of a TLS/SSL connection.
+    if (Reflect.get(socket, 'encrypted')) {
```

Comment:
Can we check the protocol instead?

Reply:
We can but that should be the discerning property to tell apart Socket from TLSSocket, per Node.js docs:

> *This may be used to distinguish TLS sockets from regular `net.Socket` instances*.
> — [Node.js docs / TLS](https://nodejs.org/docs/latest/api/tls.html#tlssocketencrypted)

I don't mind checking the protocol but if the HTTPS request does have a TLSSocket, it's constructed incorrectly. 

Is there a particular test case that fails with this check?

Reply:
Oh, it's a weird choice by Node. But if it's official, I'm in.